crates/topos/tests/push-certificate.rs

```diff
@@ -43,31 +43,35 @@ async fn assert_delivery() -> Result<(), Box<dyn std::error::Error>> {
 
     assert!(status.iter().all(|s| *s));
 
-    let nodes: String = peers_context
+    let nodes = peers_context
         .iter()
         .map(|peer| peer.1.api_entrypoint.clone())
-        .collect::<Vec<_>>()
-        .join(",");
-
-    let (tx, rx) = tokio::sync::oneshot::channel::<()>();
-
-    _ = thread::spawn(|| {
-        let mut cmd = Command::cargo_bin("topos").unwrap();
-        cmd.env("TOPOS_LOG_FORMAT", "json");
-        cmd.env("RUST_LOG", "topos=debug");
-
-        cmd.arg("tce")
-            .arg("push-certificates")
-            .args(["-f", "plain"])
-            .arg("-n")
-            .arg(nodes);
-
-        cmd.assert().success();
-
-        tx.send(()).unwrap();
-    });
-
-    _ = tokio::time::timeout(Duration::from_secs(15), rx)
+        .collect::<Vec<_>>();
+
+    info!("Nodes used in test: {:?}", nodes);
+
+    let assertion = async move {
+        let peers: Vec<tonic::transport::Uri> = nodes
+            .into_iter()
+            .map(TryInto::try_into)
+            .collect::<Result<_, _>>()
+            .map_err(|e| format!("Unable to parse node list: {e}"))
+            .expect("Valid node list");
+
+        match topos_test_sdk::integration::check_certificate_delivery(5, peers, 30).await {
```

Comment:
If you only want to test the `push_certificate` service from `topos` then you need to put this code close to it.
Either in `crates/topos/src/components/tce/services/push_certificate/tests.rs` or directly in the same file as the service.

Reply:
Ok, it make sense to do that.